client/src/hooks/useListTasks.js

```diff
@@ -0,0 +1,37 @@
+import { useTasksAndLists } from "./useTasksAndLists"
+
+// This custom hook make it easier to manage a single list.
+export const useListTasks = (listId) => {
+  // Use context data
+  const { addListTask, deleteTask, lists, tasks } = useTasksAndLists()
+
+  // Get one list
+  const list = lists[listId]
+  // TODO: add a useEffect here to redirect user if list is undefined
+  // this can by using 2 other hooks inside this one (one from React, one from react-router-dom)
+  // currently an error is thrown when this is undefined, because we try to added list.tasks below
```

Comment:
This would be a good change to make. If we could do this, we would avoid the need to call `getTitle` etc below, and check for `list` and `list.title` etc.

Reply:
If possible it would be good to get a brief conceptual overview of what hooks are :)

Reply:
Sure! Let's chat about that today.

In short, custom hooks are just like regular functions, but they can access React state/context/effects inside them. Normal functions can't do that. Because of this, React enforces some restrictions on how they can be used.

Let's chat about it when we meet.